sotodlib/core/flagman.py

```diff
@@ -0,0 +1,163 @@
+        
+        if axis_map is None:
+            try:
+                s = data.shape
+            except:
+                ### catches if a detector mask is just a list
+                s = np.shape(data)
+            if len(s) == 1:
+                if s[0] == self.dets.count:
+                    ## detector only flag. Turn into RangesMatrix
+                    axis_map=[(0,'dets')]
+                elif s[0] == self.samps.count:
+                    axis_map=[(0,'samps')]
+                else:
+                    raise ValueError("FlagManager only takes data aligned with"
+                                     " dets and/or samps. Data of shape {}"
+                                     " is the wrong shape".format(s))
+            elif len(s)==2:
+                if s[0] == self.dets.count and s[1] == self.samps.count:
+                    axis_map=[(0,'dets'), (1,'samps')]
+                elif s[1] == self.dets.count and s[0] == self.samps.count:
+                    axis_map=[(0,'samps'),(1,'dets')]
+                else:
+                    raise ValueError("FlagManager only takes data aligned with"
+                                     " dets and/or samps. Data of shape {}"
+                                     " is the wrong shape".format(s))
+            else:
+                raise ValueError("FlagManager only takes data aligned with"
+                                     " dets and/or samps. Data of shape {}"
+                                     " is the wrong shape".format(s))
+        
+        if len(axis_map)==1 and axis_map[0][1]=='dets':
+            ### Change detector flags to RangesMatrix in the backend
+            x = Ranges(self.samps.count)
+            data = RangesMatrix([Ranges.ones_like(x) if Y 
+                                 else Ranges.zeros_like(x) for Y in data])
+            axis_map = [(0,'dets'),(1,'samps')]
+        super().wrap(name, data, axis_map, *kwargs)
+                    
+    def buffer(self, n_buffer, flags=None):
+        """Buffer all the samps cuts by n_buffer
+        Like with Ranges / Ranges Matrix, buffer changes everything in place        
+        
+        Arguments:
+        
+            n_buffer: number of samples to buffer the samps cuts
+            flags: List of flags to buffer. Uses their names
+        """
+        if flags is None:
+            flags = self._fields
+        
+        for f in flags:
+            self[f].buffer(n_buffer)
+        
+    def buffered(self, n_buffer, flags=None):
+        """Return new FlagManager that has all the samps cuts buffered by n_buffer
+        Like with Ranges / Ranges Matrix, buffered returns new object
+        
+        Arguments:
+        
+            n_buffer: number of samples to buffer the samps cuts
+            flags: List of flags to buffer. Uses their names
+        """
+        new = self.copy()
+        new.buffer(n_buffer, flags)
+        return new
+        
+    def collapse(self, flags=None, method='union'):
```

Comment:
I don't get a great semantic vibe from "collapse" vs. "reduce" ...  So the difference between them is only in whether the result is stored back in the FlagManager... but their names do not help me remember which of them does that (nor does the docstring, really).  I would be more partial to keyword activation.  E.g.  fm.reduce(save_as='uber_flag'), with default behavior to not save it but only return it.